### PR TITLE
documentation - TF backend (complete)

### PR DESCRIPTION
**Pull request by keunwoochoi**
- Specify return types as many as possible.
- Make them more markdown-friendly.

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -944,34 +930,89 @@ def _normalize_axis(axis, ndim):
 
 def max(x, axis=None, keepdims=False):
     '''Maximum value in a tensor.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to find maximum values.
+        keepdims: A boolean, whether to keep the dimensions or not.
+            If `keepdims` is `False`, the rank of the tensor is reduced
+            by 1 for each entry in `axis`. If `keep_dims` is `True`,
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: `keepdims`

File: keras/backend/tensorflow_backend.py [hunk 2]
```diff
@@ -944,34 +930,89 @@ def _normalize_axis(axis, ndim):
 
 def max(x, axis=None, keepdims=False):
     '''Maximum value in a tensor.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to find maximum values.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: It should be described as a single integer, for consistency with the Theano backend (although we don't enforce this on the TF side).

File: keras/backend/tensorflow_backend.py (hunk 1 above)
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Also rewrite this description assuming an integer axis.

File: keras/backend/tensorflow_backend.py [hunk 3]
```diff
@@ -944,34 +930,89 @@ def _normalize_axis(axis, ndim):
...
     axis = _normalize_axis(axis, ndim(x))
     return tf.reduce_max(x, reduction_indices=axis, keep_dims=keepdims)
 
 
 def min(x, axis=None, keepdims=False):
     '''Minimum value in a tensor.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to find minimum values.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Same thing.

File: keras/backend/tensorflow_backend.py [hunk 4]
```diff
@@ -944,34 +930,89 @@ def _normalize_axis(axis, ndim):
...
     axis = _normalize_axis(axis, ndim(x))
     return tf.reduce_min(x, reduction_indices=axis, keep_dims=keepdims)
 
 
 def sum(x, axis=None, keepdims=False):
     '''Sum of the values in a tensor, alongside the specified axis.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to sum over.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Same thing.

File: keras/backend/tensorflow_backend.py [hunk 5]
```diff
@@ -944,34 +930,89 @@ def _normalize_axis(axis, ndim):
...
     axis = _normalize_axis(axis, ndim(x))
     return tf.reduce_sum(x, reduction_indices=axis, keep_dims=keepdims)
 
 
 def prod(x, axis=None, keepdims=False):
     '''Multiplies the values in a tensor, alongside the specified axis.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to compute the product
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Same

File: keras/backend/tensorflow_backend.py [hunk 6]
```diff
@@ -944,34 +930,89 @@ def _normalize_axis(axis, ndim):
...
     axis = _normalize_axis(axis, ndim(x))
     return tf.reduce_prod(x, reduction_indices=axis, keep_dims=keepdims)
 
 
 def var(x, axis=None, keepdims=False):
     '''Variance of a tensor, alongside the specified axis.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to compute the variance.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Same

File: keras/backend/tensorflow_backend.py [hunk 7]
```diff
@@ -985,12 +1026,34 @@ def var(x, axis=None, keepdims=False):
 
 def std(x, axis=None, keepdims=False):
     '''Standard deviation of a tensor, alongside the specified axis.
+
+    # Arguments
+        x: A tensor or variable.
+        axis: A list of integer. Axes to compute the standard deviation.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Same

**Comment by keunwoochoi** [comment]
All done!

**Comment by keunwoochoi** [comment]
@fchollet Is this good to go?

File: keras/backend/tensorflow_backend.py [hunk 8]
```diff
@@ -1339,38 +1488,67 @@ def arange(start, stop=None, step=1, dtype='int32'):
...
     if isinstance(n, int):
         n = [n]
     return tf.tile(x, n)
 
 
 def flatten(x):
+    '''Flatten a tensor.
+
+    # Returns
+        A tensor, reshaped into 1-D
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: "reshaped to 2D".

Comment (keunwoochoi) [inline keras/backend/tensorflow_backend.py]: Really? 
```
>>> import keras.backend as K
Using TensorFlow backend.
>>> a = K.ones((3, 4, 5))
>>> K.flatten(a)
<tf.Tensor 'Reshape_1:0' shape=(60,) dtype=float32>
```

Isn't it `K.batch_flatten()` that reshapes input to 2D?

Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Right. My mistake.

**Comment by fchollet** [comment]
Seems like some merge conflicts have cropped up. Please rebase and I'll merge this PR.

**Comment by keunwoochoi** [comment]
I think it's ready now.

**Comment by fchollet** [comment]
Thanks!
